File: Kattis/TrainPassengers/Train.java
import java.util.*;

public class Train {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);

    long capacity = in.nextLong();
    long numStops = in.nextLong();
    long current = 0;
    long numWait = 0;

    for (int i = 0; i < numStops; i++) {
      boolean possible = true;

      current -= in.nextLong();
      if (current < 0) possible = false;

      current += in.nextLong();
      if (current > capacity) possible = false;

      numWait = in.nextLong();
      if (current < capacity && numWait != 0) possible = false;

      if (!possible) {
        System.out.println("impossible");
        return;
      }

    }

    if (current != 0 || numWait != 0) {
      System.out.println("impossible");
    } else {
      System.out.println("possible");
    }

  }
}

File: ACM/TrainPassengers/train.py
# Train cannot exceed capacity or be below 0
# Should start and finish empty
# No passenger should wait if train is not full
# No passengers should wait at the last station

capacity, num_stops = [int(e) for e in input().split()]

# Format: [Num left] [Num entered] [Num waited]
passengers = 0

for i in range(num_stops):
    left, entered, waited = [int(e) for e in input().split()]

    passengers += entered - left
    if passengers < 0 or passengers > capacity:
        print('impossible')
        break

    if i == 0 and left != 0:
        print('impossible')
        break

    if waited != 0 and passengers != capacity:
        print('impossible')
        break

    if i == num_stops - 1:
        if passengers != 0 or waited != 0:
            print('impossible')
            break

    if i == num_stops - 1:
        print('possible')

